src/stdio_mgr/stdio_mgr.py

```diff
@@ -270,34 +274,56 @@ def stdio_mgr(in_str="", close=True):
         initially empty.
 
     """
-    if close:
-        out_cls = SafeCloseRandomTextIO
-        in_cls = SafeCloseTeeStdin
-    else:
-        out_cls = RandomTextIO
-        in_cls = TeeStdin
-
-    old_stdin = sys.stdin
-    old_stdout = sys.stdout
-    old_stderr = sys.stderr
-
-    with ExitStack() as stack:
-        new_stdout = stack.enter_context(out_cls())
-        new_stderr = stack.enter_context(out_cls())
-        new_stdin = stack.enter_context(in_cls(new_stdout, in_str))
-
-        close_files = stack.pop_all().close
-
-    sys.stdin = new_stdin
-    sys.stdout = new_stdout
-    sys.stderr = new_stderr
-
-    try:
-        yield new_stdin, new_stdout, new_stderr
-    finally:
-        sys.stdin = old_stdin
-        sys.stdout = old_stdout
-        sys.stderr = old_stderr
 
+    def __new__(cls, in_str="", close=True):
+        """Instantiate new context manager that is also a tuple."""
         if close:
-            close_files()
+            out_cls = SafeCloseRandomTextIO
+            in_cls = SafeCloseTeeStdin
+        else:
+            out_cls = RandomTextIO
+            in_cls = TeeStdin
+
+        stdout = out_cls()
+        stderr = out_cls()
+        stdin = in_cls(stdout, in_str)
+
+        self = super(stdio_mgr, cls).__new__(cls, [stdin, stdout, stderr])
```

Comment:
I could potentially achieve the same result by using https://bitbucket.org/ericvsmith/recordtype/src/default/recordtype.py or https://bitbucket.org/ericvsmith/namedlist/src/default/namedlist.py , both on PyPI.  I havent attempted that, and havent checked that they would allow `_close` to be stored on the class.  `namedtuple` definitely doesnt permit a `_close` attribute to be added.

Reply:
I'd definitely rather not add a new dependency. I'm already sort of regretting having put attrs in, esp. since it's CalVer... would be nice to have stdio-mgr be stdlib-only.

Maybe a sufficient adaptation to mimic `namedtuple` behavior would be to add `@property`s to the class definition that would wrap the `self[i]`s? As in:

```
@property
def stdin(self):
    return self[0]
```

Reply:
nice touch. Will do.